packages/flutter/lib/src/cupertino/scrollbar.dart

```diff
@@ -321,275 +174,63 @@ class _CupertinoScrollbarState extends State<CupertinoScrollbar> with TickerProv
-
-    // Convert primaryDelta, the amount that the scrollbar moved since the last
-    // time _dragScrollbar was called, into the coordinate space of the scroll
-    // position, and create/update the drag event with that position.
-    final double scrollOffsetLocal = _painter!.getTrackToScroll(primaryDelta);
-    final double scrollOffsetGlobal = scrollOffsetLocal + _currentController!.position.pixels;
-    final Axis direction = _currentController!.position.axis;
-
-    if (_drag == null) {
-      _drag = _currentController!.position.drag(
-        DragStartDetails(
-          globalPosition: direction == Axis.vertical
-            ? Offset(0.0, scrollOffsetGlobal)
-            : Offset(scrollOffsetGlobal, 0.0),
-        ),
-        () {},
-      );
-    } else {
-      _drag!.update(DragUpdateDetails(
-        globalPosition: direction == Axis.vertical
-          ? Offset(0.0, scrollOffsetGlobal)
-          : Offset(scrollOffsetGlobal, 0.0),
-        delta: direction == Axis.vertical
-          ? Offset(0.0, -scrollOffsetLocal)
-          : Offset(-scrollOffsetLocal, 0.0),
-        primaryDelta: -scrollOffsetLocal,
-      ));
-    }
-  }
-
-  void _startFadeoutTimer() {
-    if (!widget.isAlwaysShown) {
-      _fadeoutTimer?.cancel();
-      _fadeoutTimer = Timer(_kScrollbarTimeToFade, () {
-        _fadeoutAnimationController.reverse();
-        _fadeoutTimer = null;
-      });
-    }
-  }
-
-  Axis? _getDirection() {
-    try {
-      return _currentController!.position.axis;
-    } catch (_) {
-      // Ignore the gesture if we cannot determine the direction.
-      return null;
-    }
-  }
-
   double _pressStartAxisPosition = 0.0;
 
   // Long press event callbacks handle the gesture where the user long presses
   // on the scrollbar thumb and then drags the scrollbar without releasing.
-  void _handleLongPressStart(LongPressStartDetails details) {
-    _currentController = _controller;
-    final Axis? direction = _getDirection();
-    if (direction == null) {
-      return;
-    }
-    _fadeoutTimer?.cancel();
-    _fadeoutAnimationController.forward();
+
+  @override
+  void handleLongPressStart(LongPressStartDetails details) {
+    super.handleLongPressStart(details);
+    final Axis direction = getDirection()!;
     switch (direction) {
       case Axis.vertical:
         _pressStartAxisPosition = details.localPosition.dy;
-        _dragScrollbar(details.localPosition.dy);
-        _dragScrollbarAxisPosition = details.localPosition.dy;
         break;
       case Axis.horizontal:
         _pressStartAxisPosition = details.localPosition.dx;
-        _dragScrollbar(details.localPosition.dx);
-        _dragScrollbarAxisPosition = details.localPosition.dx;
         break;
     }
   }
 
-  void _handleLongPress() {
-    if (_getDirection() == null) {
+  @override
+  void handleLongPress() {
+    if (getDirection() == null) {
       return;
     }
-    _fadeoutTimer?.cancel();
+    super.handleLongPress();
```

Comment:
If subclasses are allowed to opt out of the super call, the base class will have trouble reasoning about how the handleFoo methods update its state.

Reply:
I should include a `@mustCallSuper`